From balbi@ti.com  Tue Oct  5 13:30:50 2010
From: Felipe Balbi <balbi@ti.com>
To: Greg KH <greg@kroah.com>
Cc: Linux USB Mailing List <linux-usb@vger.kernel.org>,
        Heikki Krogerus <ext-heikki.krogerus@nokia.com>
Subject: usb: musb: ignore spurious SESSREQ interrupts
Date: Fri, 24 Sep 2010 13:44:13 +0300
Message-Id: <1285325055-1247-13-git-send-email-balbi@ti.com>

From: Heikki Krogerus <ext-heikki.krogerus@nokia.com>

This will ignore any SESSREQ interrupt if musb is B state.
Charger detection may cause spurious SESSREQ interrupts.

Signed-off-by: Heikki Krogerus <ext-heikki.krogerus@nokia.com>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>
---
 drivers/usb/musb/musb_core.c |    5 +++++
 1 file changed, 5 insertions(+)

--- a/drivers/usb/musb/musb_core.c
+++ b/drivers/usb/musb/musb_core.c
@@ -550,6 +550,11 @@ static irqreturn_t musb_stage0_irq(struc
 	if (int_usb & MUSB_INTR_SESSREQ) {
 		void __iomem *mbase = musb->mregs;
 
+		if (devctl & MUSB_DEVCTL_BDEVICE) {
+			DBG(3, "SessReq while on B state\n");
+			return IRQ_HANDLED;
+		}
+
 		DBG(1, "SESSION_REQUEST (%s)\n", otg_state_string(musb));
 
 		/* IRQ arrives from ID pin sense or (later, if VBUS power
